index.js

```diff
@@ -309,15 +309,12 @@ Kareem.prototype.wrap = function(name, fn, context, args, options) {
           argsWithoutError, options, lastArg);
       } else {
         _this.execPost(name, context, argsWithoutError, function() {
-          if (arguments[0]) {
-            return typeof lastArg === 'function' ?
-              lastArg(arguments[0]) :
-              undefined;
+          if (lastArg === null) {
+            return;
           }
-
-          return typeof lastArg === 'function' ?
-            lastArg.apply(context, arguments) :
-            undefined;
+          arguments[0]
```

Comment:
Shouldn't there be a `return` here? Also, don't you still need the `typeof lastArg === 'function'` check?

Reply:
lastArg is only defined when the Last arg is of Type function.  See line 259 were we initialize and set lastArg